src/shared/components/search.tsx

```diff
@@ -64,51 +67,36 @@ import { HtmlTags } from "./common/html-tags";
 import { Spinner } from "./common/icon";
 import { ListingTypeSelect } from "./common/listing-type-select";
 import { Paginator } from "./common/paginator";
+import { SearchableSelect } from "./common/searchable-select";
 import { SortSelect } from "./common/sort-select";
 import { CommunityLink } from "./community/community-link";
 import { PersonListing } from "./person/person-listing";
 import { PostListing } from "./post/post-listing";
 
-var Choices: any;
-if (isBrowser()) {
-  Choices = require("choices.js");
-}
-
 interface SearchProps {
   q?: string;
-  type_: SearchType;
+  type: SearchType;
   sort: SortType;
   listingType: ListingType;
-  communityId: number;
-  creatorId: number;
+  communityId?: number | null;
```

Comment:
I made these nullable because sometimes they need to be explicitly set to nothing instead of just undefined.

Reply:
Perfect.